### PR TITLE
Continue to clean up HTMLMediaElement

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18520)

### CODE REVIEW

File: components/script/dom/htmlmediaelement.rs [hunk 1]
```diff
@@ -230,86 +226,66 @@ impl HTMLMediaElement {
...
-            // and the new ready state is ReadyState::HaveFutureData.
-            (ReadyState::HaveCurrentData, ReadyState::HaveFutureData) |
-            (ReadyState::HaveMetadata, ReadyState::HaveFutureData) |
-            (ReadyState::HaveNothing, ReadyState::HaveFutureData) => {
-                task_source.queue_simple_event(
-                    self.upcast(),
-                    atom!("canplay"),
-                    &window,
-                );
+        if old_ready_state <= ReadyState::HaveCurrentData && ready_state >= ReadyState::HaveFutureData {
```
Comment (emilio) [inline components/script/dom/htmlmediaelement.rs]: This isn't idempotent, you weren't testing for `HaveEnoughData` before.

File: components/script/dom/htmlmediaelement.rs [hunk 2]
```diff
@@ -230,86 +226,66 @@ impl HTMLMediaElement {
...
-                        &window,
-                    );
-
-                    if !self.Paused() {
-                        self.notify_about_playing();
-                    }
-                }
+        if ready_state != ReadyState::HaveEnoughData {
+            // TODO: Check sandboxed automatic features browsing context flag.
+            // FIXME(nox): I have no idea what this TODO is about.
```
Comment (emilio) [inline components/script/dom/htmlmediaelement.rs]: lol

**Review by emilio — APPROVED** [review]
That fixed or double-checked, and r=me

File: components/script/dom/htmlmediaelement.rs (hunk 1 above)
Comment (nox) [reply components/script/dom/htmlmediaelement.rs]: This transition should always be done, and then the final one to `HaveEnoughData`.

**Comment by mbrubeck** [comment]
@bors-servo

infra

**Comment by nox** [comment]
Oh I forgot about this…

**Comment by nox** [comment]
This fixes a typo in the very first commit. Should be ok now.

@bors-servo r=emilio

**Comment by nox** [comment]
Turns out "canplaythrough" must come after "play", as opposed to what the spec says. I will file an issue against the spec.

@bors-servo r=emilio

**Pull request merged**